shell.nix

```diff
@@ -57,15 +95,22 @@ let inherit (nixroot) stdenv pkgs lib
     fi
   '';
 
+  wrapper = (if builtins.hasAttr "wrapQtAppsHook" pkgs then
+    "source ${pkgs.wrapQtAppsHook}/nix-support/setup-hook;\nwrapQtApp"
+  else
+    "source ${pkgs.makeWrapper}/nix-support/setup-hook;\nwrapProgram");
+
+  wrapperCmd = (if builtins.hasAttr "wrapQtAppsHook" pkgs then "wrapQtApp" else "wrapProgram");
+
   shell-build = nixroot.writeShellScriptBin "build" ''
     if [ ! -d "cbuild" ]; then
       >&2 echo "First you have to run configure."
       exit 1
     fi
     cd cbuild
+    rm -f .mixxx-wrapped mixxx
     cmake --build . --parallel $NIX_BUILD_CORES "$@"
-    source ${pkgs.makeWrapper}/nix-support/setup-hook
-    wrapProgram mixxx --prefix LV2_PATH : ${lib.makeSearchPath "lib/lv2" allLv2Plugins}
+    ${wrapper} mixxx --prefix LV2_PATH : ${lib.makeSearchPath "lib/lv2" allLv2Plugins}
```

Comment:
```suggestion
    source ${dirOf wrapperCmd}/../nix-support/setup-hook
    ${wrapperCmd} mixxx --prefix LV2_PATH : ${lib.makeSearchPath "lib/lv2" allLv2Plugins}
```

Reply:
thats clever

Reply:
unfortuntelly, it does not work because the wrapper is not a bin file but a function defined in the shell support hook only.
I remember trying something like this...